### PR TITLE
[UNR-4008][UNR-3902] Fix RPCs being dropped randomly

### PR DESCRIPTION
**Pull request by mironec**
#### Description
RPCs are now less likely to be dropped.

#### Release note
Not needed, should have worked.

#### Tests
We should probably have had a unit test, as this could have been caught, but hard to think of that in advance, as it's essentially identical to solving the bug outright. Can do a regression unit test as a follow-up.

#### Documentation
N/A

#### Primary reviewers
@improbable-valentyn

### CODE REVIEW

**Comment by improbable-valentyn** [comment]
Note: this doesn't need a release note because this bug was not included in the prior release.

File: SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp [hunk 1]
```diff
@@ -125,12 +125,16 @@ void FRPCContainer::ProcessRPCs(FArrayOfParams& RPCList)
...
+			break;
 		case ERPCQueueProcessResult::StopProcessing:
+			// The actual break out of the for loop is handled below
 			break;
 		case ERPCQueueProcessResult::DropEntireQueue:
 			RPCList.Empty();
 			return;
 		}
+
+		if (QueueProcessResult == ERPCQueueProcessResult::StopProcessing) break;
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp:137]:
```suggestion
		if (QueueProcessResult == ERPCQueueProcessResult::StopProcessing)
		{
			break;
		}
```

**Comment by mironec** [comment]
> Note: this doesn't need a release note because this bug was not included in the prior release.

Oh, is this new code? I thought it was in 0.10?

**Comment by improbable-valentyn** [comment]
> > Note: this doesn't need a release note because this bug was not included in the prior release.
> 
> Oh, is this new code? I thought it was in 0.10?

Yes, this is new. It used to be an if/else in 0.10:
https://github.com/spatialos/UnrealGDK/blob/0.10.0/SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp

**Pull request merged**